packages/cli/test/integration/task-runners/js-task-runner-execution.integration.test.ts

```diff
@@ -0,0 +1,264 @@
+import { TaskRunnersConfig } from '@n8n/config';
+import { Container } from '@n8n/di';
+import { mock } from 'jest-mock-extended';
+import type {
+	IExecuteFunctions,
+	INode,
+	INodeExecutionData,
+	INodeParameters,
+	INodeTypes,
+	IRunExecutionData,
+	ITaskDataConnections,
+	IWorkflowExecuteAdditionalData,
+	WorkflowExecuteMode,
+} from 'n8n-workflow';
+import { createEnvProviderState, NodeConnectionType, Workflow } from 'n8n-workflow';
+
+import { LocalTaskRequester } from '@/task-runners/task-managers/local-task-requester';
+import { TaskRunnerModule } from '@/task-runners/task-runner-module';
+
+/**
+ * Integration tests for the JS TaskRunner execution. Starts the TaskRunner
+ * as a child process and executes tasks on it via the broker.
+ */
+describe('JS TaskRunner execution', () => {
+	const runnerConfig = Container.get(TaskRunnersConfig);
+	runnerConfig.mode = 'internal';
+	runnerConfig.enabled = true;
+	runnerConfig.port = 45678;
+
+	const taskRunnerModule = Container.get(TaskRunnerModule);
+	const taskRequester = Container.get(LocalTaskRequester);
+
+	/**
+	 * Sets up task data that includes a workflow with manual trigger and a
+	 * code node with the given JS code. The input data is a single item:
+	 * ```json
+	 * {
+	 *   "input": "item"
+	 * }
+	 * ```
+	 */
+	const newTaskData = (jsCode: string) => {
+		const taskSettings = {
+			code: jsCode,
+			nodeMode: 'runOnceForAllItems',
+			workflowMode: 'manual',
+			continueOnFail: false,
+		};
+
+		const codeNode = {
+			parameters: {
+				jsCode,
+			},
+			type: 'n8n-nodes-base.code',
+			typeVersion: 2,
+			position: [200, 80],
+			id: 'b35fd455-32e4-4d52-b840-36aa28dd1910',
+			name: 'Code',
+		} satisfies INode;
```

Comment:
Why not a type annotation? I don't think this fits the use cases for `satisfies`

Reply:
Good point. I wanted to keep it explicitly with type code node, but it doesn't seem to matter really. Changed to type